service/gcs/bridge/bridge.go

```diff
@@ -34,6 +34,11 @@ type bridge struct {
 	// message data is kept together when writing from multiple go routines
 	// simultaneously.
 	writeLock sync.Mutex
+
+	// killed and killedLock determine if the bridge has been killed.
+	// killedLock must be held when reading or writing killed.
+	killed     bool
+	killedLock sync.Mutex
```

Comment:
I considered that, but preferred the strong typing of bools over ints. That said, I don't really have a strong opinion either way and I think atomics are more performant at least in C++, so I'll make the change :)